Add global_level for analyzer config (#25034)
dotnet/docs: docs/fundamentals/code-analysis/configuration-files.md

### Naming

Unlike EditorConfig files, which must be named `.editorconfig`, global config files do not need to have a specific name or extension. However, if you name these files as `.globalconfig`, then they are implicitly applied to all the C# and Visual Basic projects within the current folder, including subfolders. Otherwise, you must explicitly add the `GlobalAnalyzerConfigFiles` item to your MSBuild project file:

```xml
<ItemGroup>
  <GlobalAnalyzerConfigFiles Include="<path_to_global_analyzer_config>" />
</ItemGroup>
```

Consider the following naming recommendations:

> - End users should name their global config files *.globalconfig*.

Reviewer: You changed "config" to "configuration" earlier..

```suggestion
- End users should name their global configuration files *.globalconfig*.
```

> - NuGet packages creators should name their global config files *<%Package_Name%>.globalconfig*.

Reviewer: ```suggestion
- NuGet package creators should name their global config files *<%Package_Name%>.globalconfig*.
```

Normally you'd have to escape `<` but it's working OK this way -- I don't know why.

Reviewer 2: > Normally you'd have to escape < but it's working OK this way -- I don't know why.

It *may* not render correctly for all parsers.

| In the same configuration file | The entry that appears later in the file wins. This is true for conflicting entries within a single EditorConfig file and also within a single global AnalyzerConfig file. |
| In two EditorConfig files | The entry in the EditorConfig file that's deeper in the file system, and hence has a longer file path, wins. |
| In two global AnalyzerConfig files | **.NET 5**: A compiler warning is reported and both entries are ignored.<br/>**.NET 6 and later versions**: The entry from the file with a higher value for `global_level` takes precedence. If `global_level` isn't explicitly defined and the file is named *.globalconfig*, the `global_level` value defaults to `100`; for all other global AnalyzerConfig files, `global_level` defaults to `0`. If the `global_level` values for the configuration files with conflicting entries are equal, a compiler warning is reported and both entries are ignored. |
| In an EditorConfig file and a Global AnalyzerConfig file | The entry in the EditorConfig file wins. |

### Severity options

For [severity configuration](configuration-options.md#severity-level) options, the following *additional* precedence rules apply:

> - Severity options specified at the command line as compiler options (`/nowarn` or `/warnaserror`) always override [severity configuration](configuration-options.md#severity-level) options specified in EditorConfig and global AnalyzerConfig files.

Reviewer 2: Unrelated to the changes being made here. See https://github.com/dotnet/docs/issues/4072.

```suggestion
- Severity options specified at the command line as compiler options (`-nowarn` or `-warnaserror`) always override [severity configuration](configuration-options.md#severity-level) options specified in EditorConfig and global AnalyzerConfig files.
```

- The precedence rules for conflicting severity entries from a [ruleset file](/visualstudio/code-quality/using-rule-sets-to-group-code-analysis-rules) and an EditorConfig or global AnalyzerConfig file is _undefined_.

>   > Ruleset files are deprecated in favor of EditorConfig and global AnalyzerConfig files. It's recommended that you [convert ruleset files to an equivalent EditorConfig file](/visualstudio/code-quality/use-roslyn-analyzers#convert-an-existing-ruleset-file-to-editorconfig-file).

Reviewer: ```suggestion
  > Ruleset files are deprecated in favor of EditorConfig and global AnalyzerConfig files. We recommend that you [convert ruleset files to an equivalent EditorConfig file](/visualstudio/code-quality/use-roslyn-analyzers#convert-an-existing-ruleset-file-to-editorconfig-file).
```

Reviewer 2: Shouldn't we be encouraging moving from rulesets to globalconfigs, instead of editorconfigs?

Author: I'm not sure. @mavasani ?
